modules/logging/ui/src/sap/commerce/toolset/logging/actionSystem/CxCreateCustomTemplateStateAction.kt

```diff
@@ -0,0 +1,174 @@
+        else -> "Create a Log Template"
+    }
+
+    private fun templateNameFromConnection(project: Project, connectionName: String): String {
+        val templateName = "Remote '$connectionName' | template"
+        val count = CxCustomLogTemplatesSettings.getInstance(project).templates.count { it.name.startsWith(templateName) }
+
+        return if (count >= 1) "$templateName ($count)"
+        else templateName
+    }
+
+    private fun loggers(project: Project, selectedNodes: List<Any>) = selectedNodes
+        .mapNotNull {
+            when (it) {
+                is CxRemoteLogStateNode -> CxRemoteLogStateService.getInstance(project).state(it.connection.uuid).get()?.values ?: emptyList()
+                is CxBundledLogTemplateItemNode -> it.loggers
+                is CxCustomLogTemplateItemNode -> it.loggers
+                else -> null
+            }
+        }
+        .flatten()
+
+    private fun templateName(project: Project, selectedNodes: List<Any>) = when (val firstNode = selectedNodes.first()) {
+        is CxRemoteLogStateNode -> templateNameFromConnection(project, firstNode.connection.shortenConnectionName)
+        is CxBundledLogTemplateItemNode -> if (selectedNodes.size > 1) "Merged templates" else "Copy of '${firstNode.name}'"
+        is CxCustomLogTemplateItemNode -> if (selectedNodes.size > 1) "Merged templates" else "Copy of '${firstNode.name}'"
+        else -> ""
+    }
+
+    override fun update(e: AnActionEvent) {
+        e.presentation.isVisible = ActionPlaces.ACTION_SEARCH != e.place
+        val selectedNodes = e.selectedNodes() ?: return
+        val project = e.project ?: return
+
+        e.presentation.isEnabled = isActionEnabled(project, selectedNodes)
+
+        if (!e.presentation.isVisible) return
+
+        e.presentation.text = actionLabel(selectedNodes)
+        e.presentation.icon = HybrisIcons.Log.Action.SAVE_AS_TEMPLATE
+        e.presentation.disabledIcon = disabledIcon(project, selectedNodes)
+    }
+
+    private fun isActionEnabled(project: Project, selectedNodes: List<Any>): Boolean {
+        //if incompatible nodes selected => disable the action
+        if (selectedNodes.groupBy { it.javaClass }.keys.size > 1) return false
+
+        val loggerAccess = CxRemoteLogStateService.getInstance(project)
+
+        return when (val firstNode = selectedNodes.first()) {
+            //if remote state is fetched ==> enable the action
+            is CxRemoteLogStateNode -> loggerAccess.ready
+                && loggerAccess.stateInitialized
+                && loggerAccess.state(firstNode.connection.uuid).get()?.isNotEmpty() ?: false
+
+            is CxBundledLogTemplateItemNode -> true
+            is CxCustomLogTemplateItemNode -> true
+            else -> false
+        }
+    }
+
+    private fun disabledIcon(project: Project, selectedNodes: List<Any>) = when (selectedNodes.first()) {
```

Comment:
replace `selectedNodes.first()` with the parameter `selectedNode`.
`first()` may throw NPE, is it possible in this case?